android/src/main/java/com/revenuecat/purchases/hybridcommon/mappers/StoreProductMapper.kt

```diff
@@ -102,28 +102,31 @@ internal fun StoreProduct.mapIntroPrice(): Map<String, Any?>? {
     }
 }
 
-private fun Period.mapPeriod(): Map<String, Any?>? {
+private fun Period.mapPeriod(forSubscriptionOption: Boolean): Map<String, Any?>? {
+    val unitKey = if (forSubscriptionOption) "unit" else "periodUnit";
+    val valueKey = if (forSubscriptionOption) "value" else "periodNumberOfUnits"
```

Comment:
Hmm I think changing the mapping for the same object can be confusing... 

If we think having `unit` and `value` is enough info and considering when we release this, it will likely be a major, I think we should just rename it everywhere. If this is not released as a major, we could deprecate the old fields and keep them around. Thoughts?